lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,327 @@
+        self.pl_dict = problightning_values
+
+        if precip_method is None:
+            self.precip_plugin = ApplyPrecip(problightning_values=self.pl_dict)
+        else:
+            self.precip_plugin = precip_method
+        self.ice_plugin = ice_method
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius}, debug={debug},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+With:
+{precip}
+{ice}
+>""".format(radius=self.radius, debug=self.debug,
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2],
+            precip=self.precip_plugin,
+            ice=self.ice_plugin)
+
+    def _process_haloes(self, cube):
+        """
+        Adjust data so that lightning probability does not decrease too rapidly
+        with distance.
+
+        Args:
+            cube (iris.cube.Cube):
+                Radius will be applied equally on x and y dimensions.
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube of same shape as cube with haloes applied.
+        """
+        new_cube = self.neighbourhood.process(cube.copy())
+        return new_cube
+
+    def _update_meta(self, cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
+        new_cube.rename("lightning_probability")
+        new_cube.remove_coord('threshold')
+        new_cube.attributes = {}
+        if self.debug:
+            print(('In {}, new_cube is {}'.format(self, new_cube)))
+        return new_cube
+
+    def _modify_first_guess(self, cube, fg_cube, ltng_cube, precip_cube,
+                            vii_cube):
+        """
+        Modify first-guess lightning probability with nowcast data
+
+        Args:
+            cube (iris.cube.Cube):
+                Provides the meta-data for the Nowcast lightning probability
+                output cube
+
+            fg_cube (iris.cube.Cube):
+                First-guess lightning probability
+                Must have same x & y dimensions as cube
+                Time dimension should overlap that of cube
+
+            ltng_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
```

Comment:
Please could you expand these cube names to be more explicit.  The code below is very difficult to read.